lib/kubernetes-deploy/resource_watcher.rb

```diff
@@ -48,6 +45,18 @@ def run(delay_sync: 3.seconds, reminder_interval: 30.seconds, record_summary: tr
 
     private
 
+    def global_timeout?(started_at)
+      @timeout && (Time.now.utc - started_at > @timeout)
+    end
+
+    def sleep_until_next_sync(min_interval)
+      @next_sync_time ||= Time.now.utc
+      if (sleep_duration = @next_sync_time - Time.now.utc) > 0
+        sleep(sleep_duration)
+      end
+      @next_sync_time = Time.now.utc + min_interval
```

Comment:
I'm scratching my head trying to understand whether or not this line should be in the `if` branch above, only in the `else` branch, or here where it is. Do we want to bump the `next_sync_time` even if we didn't actually sleep? It's a bit too late in the day here for me to have enough mental clarity to say, but maybe this can be expressed more clearly somehow?

Reply:
I just saw that's the previous behaviour already, so don't let my comment block you...

Reply:
This code is pretty much a really dumb throttle. It assumes it is called before every sync and its job is to make sure the syncs begin at least `min_interval` seconds apart. So yes, we need to set the `next_sync_time` every time we're about to leave the method, i.e start a sync. Any suggestions on naming? Were the previous names clearer?